From d2f0c30b524cdd310cd8f8aa6a788bf26d4ccdc2 Mon Sep 17 00:00:00 2001
From: Sinthu Raja <sinthu.raja@ti.com>
Date: Thu, 16 Feb 2023 12:13:25 +0530
Subject: [PATCH 1/2] board: ti: j721s2: enable probing daughter cards.

Add support to detect the board revision of the AM68 SK board. This
idnetifies the board revision and set the board_rev env variable to
select the board revision specific dtb file for kernel boot.`

Signed-off-by: Sinthu Raja <sinthu.raja@ti.com>
---
 board/ti/j721s2/evm.c | 13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/board/ti/j721s2/evm.c b/board/ti/j721s2/evm.c
index 1227a5913f..559f924e1e 100644
--- a/board/ti/j721s2/evm.c
+++ b/board/ti/j721s2/evm.c
@@ -190,6 +190,12 @@ static const struct {
 		"k3-j721s2-gesi-exp-board.dtbo",
 		1,		/* Start populating from eth1addr */
 	},
+	{
+		0x52,
+		"AM68-SK-BB",
+		"",
+		0,
+	},
 };
 
 #define DAUGHTER_CARD_NO_OF_MAC_ADDR	5
@@ -276,6 +282,11 @@ static int probe_daughtercards(void)
 			if (!daughter_card_detect_flags[i])
 				continue;
 
+			if (!strncmp(ep.name,"AM68-SK-BB",10))  {
+				env_set("board_ver", ep.version);
+				continue;
+			}
+
 			/* Skip if no overlays are to be added */
 			if (!strlen(ext_cards[i].dtbo_name))
 				continue;
@@ -330,7 +341,7 @@ int board_late_init(void)
 	if (IS_ENABLED(CONFIG_TI_I2C_BOARD_DETECT)) {
 		setup_board_eeprom_env();
 		setup_serial();
-		if (board_is_j721s2_som())
+		//if (board_is_j721s2_som()) /* Commenting this check for E1 compatibility */
 			probe_daughtercards();
 	}
 
-- 
2.36.1

